doc/specs/#2046 - Command Palette.md

```diff
@@ -401,6 +484,184 @@ The `{ "key": "resourceName" }` solution proposed here was also touched on in
+          },
+          {
+            "icon": "...",
+            "name": { "key": "SplitPaneVerticalName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "vertical" }
+          },
+          {
+            "icon": "...",
+            "name": { "key": "SplitPaneHorizontalName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "horizontal" }
+          }
+        ]
+      }
+    ]
+  }
+]
+```
+
+This would define three commands, each with a number of nested commands underneath it:
+* For the first command:
+  - It uses the XAML resource `NewTabWithProfileRootCommandName` as it's name.
+  - Activating this command would cause us to remove all the other commands from
+    the command palette, and only show the nested commands.
+  - It contains nested commands, one for each profile.
+    - Each nested command would use the XAML resource
+      `NewTabWithProfileCommandName`, which then would also contain the string
+      `${profile.name}`, to be filled with the profile's name in the command's
+      name.
+    - It would also use the profile's icon as the command icon.
+    - Activating any of the nested commands would dispatch an action to create a
+      new tab with that profile
+* The second command:
+  - It uses the string literal `"Connect to ssh..."` as it's name
+  - It contains two nested commands:
+    - Each nested command has it's own literal name
+    - Activating these commands would cause us to open a new tab with the
+      provided `commandline` instead of the default profile's `commandline`
+* The third command:
+  - It uses the XAML resource `NewTabWithProfileRootCommandName` as it's name.
+  - It contains nested commands, one for each profile.
+    - Each one of these sub-commands each contains 3 subcommands - one that will
+      create a new split pane automatically, one vertically, and one
+      horizontally, each using the given profile.
+
+So, you could imagine the entire tree as follows:
+
+```
+<Command Palette>
+├─ New Tab With Profile...
+│  ├─ Profile 1
+│  ├─ Profile 2
+│  └─ Profile 3
+├─ Connect to ssh...
+│  ├─ first.com
+│  └─ second.com
+└─ New Pane...
+   ├─ Profile 1...
+   |  ├─ Split Automatically
+   |  ├─ Split Vertically
+   |  └─ Split Horizontally
+   ├─ Profile 2...
+   |  ├─ Split Automatically
+   |  ├─ Split Vertically
+   |  └─ Split Horizontally
+   └─ Profile 3...
+      ├─ Split Automatically
+      ├─ Split Vertically
+      └─ Split Horizontally
+```
+
+### Configuring the Action/Commandline Mode prefix
+
+As always, I'm also on board with the "this should be configurable by the user"
+route, so they can change what mode the command palette is in by default, and
+what the prefixes for differen modes are, but I'm not sure how we'd define that
+cleanly in the settings.
+
+```json
+{
+  "commandPaletteActionModePrefix": "", // or null, for no prefix
+  "commandPaletteCommandlineModePrefix": ">"
+}
+```

```

Comment:
Should we do what we did with "profiles" and overload the parser to accept an object and an array?
- Rename "commands" --> "commandPalette"
- Accepted children include:
   - "actionModePrefix"
   - "commandlineModePrefix"
   - "list" (list of commands)
Then this would be easily extensible to customize other things in the future like...
- control the max number of results displayed
- theming? (i.e.: font size, position, colors, etc...)